reinforcement_learning/rlclientlib/utility/data_buffer.cc

```diff
@@ -3,50 +3,43 @@
 #include <iterator>
 #include <iostream>
 
-namespace reinforcement_learning { namespace utility {
-  data_buffer::data_buffer() = default;
+namespace reinforcement_learning {
+  namespace utility {
+    data_buffer::data_buffer() = default;
 
-  data_buffer::data_buffer(const translate_func& _translate) : translate(_translate) {}
+    void data_buffer::reset() { _buffer.clear(); }
 
-  void data_buffer::reset() { _buffer.clear(); }
+    size_t data_buffer::size() const {
+      return _buffer.size();
+    }
 
-  std::string data_buffer::str() const{
-    std::string retval(std::begin(_buffer), std::end(_buffer));
-    return retval;
-  }
-
-  size_t data_buffer::size() const {
-    return _buffer.size();
-  }
-
-  void data_buffer::remove_last() { _buffer.pop_back(); }
+    void data_buffer::remove_last() { _buffer.pop_back(); }
 
-  buffer_factory::buffer_factory(const translate_func& _translate) : translate(_translate) {}
+    void data_buffer::append(const unsigned char* data, size_t len) {
+      _buffer.reserve(_buffer.size() + len);
+      _buffer.insert(_buffer.end(), data, data + len);
+    }
 
-  data_buffer* buffer_factory::operator()() const { return new data_buffer(translate); }
+    std::vector<unsigned char> data_buffer::buffer() {
```

Comment:
Should this be a copy? Shouldn't we be returning a reference or pointer here?

Reply:
it's git hub diff confused you. The real diff of this line should be line 43.

Reply:
I'm not sure I understand? We are still returning a complete copy of the buffer here. Generally a function named liked this would return a pointer to the underlying buffer

Reply:
This is a new method. Previous method is in line 43 in right patch, leaving unchanged.

Reply:
This can be changed to get_buffer()

Reply:
I think I got what you mean - you are suggesting returning "std::vector<unsigned char>&"?

Reply:
Yeah that or just a pointer